awshelper.py

```diff
@@ -11,45 +11,69 @@ def _get_client(client_type, region):
         raise ValueError("Session has not been initialized")
     return _session.client(client_type, region_name=region)
 
-
-def _aws_response(response_func, args_dict):
-    response = response_func(**args_dict)
-    metadata = response["ResponseMetadata"]
-    if "HTTPStatusCode" not in metadata or metadata["HTTPStatusCode"] != 200:
-        raise IOError("Unexpected status code " + metadata.get("HTTPStatusCode", None))
-    response.pop("ResponseMetadata")
-    return response
+NON_PAGINATED_METHODS = [ "describe_load_balancer_attributes"
+    , "describe_addresses"
+    , "describe_availability_zones"
+    , "describe_customer_gateways"
+    , "describe_target_health"
+    , "describe_vpn_connections"
+    , "describe_vpn_gateways"
+    , "describe_elasticsearch_domains"
+    , "search_transit_gateway_routes"
+    ]
+
+def _aws_response(region, cli_name, cli_method, args_dict):
+    # See https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html
+    # also https://stackoverflow.com/questions/39201093/how-to-use-boto3-pagination on subtleties
+    client = _clients[region][cli_name]
+    can_paginate = cli_method not in NON_PAGINATED_METHODS
+    if can_paginate:
+        paginator = client.get_paginator(cli_method)
+        page_iterator = paginator.paginate(**args_dict)
+        response = page_iterator.build_full_result()
+        # Allegedly boto3 will throw an exception if it encounters an error,
+        # so no need to explicitly check.
+        return response
+    else:
+        fun = getattr(client, cli_method)
+        response = fun(**args_dict)
+        metadata = response["ResponseMetadata"]
+        if "HTTPStatusCode" not in metadata or metadata["HTTPStatusCode"] != 200:
+            raise IOError("Unexpected status code " + metadata.get("HTTPStatusCode", None))
+        response.pop("ResponseMetadata")
+        return response
 
 
 def aws_get_config(region):
+    print("Querying region: ", region)
     config = {}
     for key in _functions[region]:
-        (response_func, args_dict) = _functions[region][key]
+        (cli_name, cli_method, args_dict) = _functions[region][key]
         try:
-            response = _aws_response(response_func, args_dict)
+            response = _aws_response(region, cli_name, cli_method, args_dict)
```

Comment:
Note: what do you think @ratulm , maybe in a follow-up PR these tries could be optionally made (based on switch) fail-hard. To avoid accidentally ending up with partial data without noticing (like in a batch job or such).

Reply:
Sounds like a good idea.